fix B0003 example and update logs (#11162)
bevyengine/bevy: errors/B0003.md

    App::new()
        .add_plugins(DefaultPlugins)
        .add_systems(Startup, setup)
        .add_systems(Update, (use_0_and_despawn_1, use_1_and_despawn_0))
        .run();
}

#[derive(Resource)]

> struct MyEntities([Entity; 2]);

Reviewer: I think this would be better as 2 tuple arguments instead of an array.
```suggestion
struct MyEntities(Entity, Entity);
```

Author: using a tuple means accessing the entities from a resource with `my_entities.0 .0` which is less pretty than the array...

Reviewer: If you do:
```rs
struct MyEntities(Entity, Entity);
```
instead of:
```rs
struct MyEntities((Entity, Entity));
```
then you can do `my_entities.0` / `my_entities.1` instead of `my_entities.0.0` or `my_entities.0[0]`

Author: ha right 😄

#[derive(Component)]
struct Hello;

fn setup(mut commands: Commands) {
    let entity1 = commands.spawn_empty().id();
    let entity2 = commands.spawn_empty().id();
    commands.insert_resource(MyEntities([entity1, entity2]));
}

fn use_0_and_despawn_1(mut commands: Commands, my_entities: Res<MyEntities>) {
    commands.entity(my_entities.0[0]).insert(Hello);
    commands.entity(my_entities.0[1]).despawn();
}

fn use_1_and_despawn_0(mut commands: Commands, my_entities: Res<MyEntities>) {
    commands.entity(my_entities.0[1]).insert(Hello);
    commands.entity(my_entities.0[0]).despawn();
}
```

> This will panic, as whichever system that is executed first will despawn the entity used by the second.

Reviewer 2: ```suggestion
This will panic, as the system that is executed first will despawn the entity used by the second.
```

> The default panic message is telling you the entity id (`2v0`), the command that failed (adding a component `Hello`) and the system from which it originated (`use_1_and_despawn_0`):

Reviewer: This grammar doesn't make sense.
```suggestion
The default panic message is telling you the entity (`2v0`) doesn't exist, and the command that failed. (adding a component `Hello`) and the system from which it originated (`use_1_and_despawn_0`):
```
Additionally there are 2 cases for what entity doesn't exist (either `1v0` or `2v0` based on system ordering)

```text
thread 'main' panicked at /bevy/crates/bevy_ecs/src/system/commands/mod.rs:1097:13:
error[B0003]: Could not insert a bundle (of type `use_entity_after_despawn::Hello`) for entity 2v0 because it doesn't exist in this World.
Encountered a panic when applying buffers for system `use_entity_after_despawn::use_1_and_despawn_0`!
Encountered a panic in system `bevy_app::main_schedule::Main::run_main`!
```

> To get the system that created the despawn command, you can enable DEBUG logs for crate `bevy_ecs`, for example by setting the environment variable `RUST_LOG=bevy_ecs=debug`. This will log:

Reviewer: Maybe mention that you can also configure `LogPlugin` to show debug logs?
